File: Mondrian/src/main/java/hcs/offloading/mondrian/BoundingBox.java
package hcs.offloading.mondrian;

public class BoundingBox {
    public final int id;
    public final int left;
    public final int top;
    public final int right;
    public final int bottom;
    public final float confidence;
    public final int label;
    public final int origin;
    public final boolean isNew;

    public BoundingBox(int id, int left, int top, int right, int bottom, float confidence, int label, int origin, boolean isNew) {
        this.id = id;
        this.left = left;
        this.top = top;
        this.right = right;
        this.bottom = bottom;
        this.confidence = confidence;
        this.label = label;
        this.origin = origin;
        this.isNew = isNew;
    }
}

File: android/Mondrian/src/main/cpp/mondrian/src/BoundingBox.cpp
#include "mondrian/BoundingBox.hpp"

#include "mondrian/ROI.hpp"

namespace md {

BoundingBox::BoundingBox(OID oid,
                         PID pid,
                         const cv::Rect2f& loc,
                         const float confidence,
                         int label,
                         BoxOrigin origin)
    : bid(BoundingBox::nextBID_++),
      oid(oid),
      pid(pid),
      loc(loc),
      confidence(confidence),
      label(label),
      origin(origin),
      srcROI_(nullptr),
      srcROIID_(-1),
      choiceOfBox(INVALID_OID) {}

void BoundingBox::setSrcROI(ROI* roi) {
  srcROI_ = roi;
  srcROIID_ = roi != nullptr ? roi->rid : -1;
}

std::string BoundingBox::header() {
  std::stringstream ss;
  ss << "bid" << DELIM
     << "oid" << DELIM
     << "pid" << DELIM
     << "box_l" << DELIM
     << "box_t" << DELIM
     << "box_r" << DELIM
     << "box_b" << DELIM
     << "confidence" << DELIM
     << "label" << DELIM
     << "srcROI" << DELIM
     << "origin" << DELIM
     << "choiceOfBox";
  return ss.str();
}

std::string BoundingBox::str() const {
  std::stringstream ss;
  ss << bid << DELIM
     << oid << DELIM
     << pid << DELIM
     << loc.x << DELIM
     << loc.y << DELIM
     << loc.x + loc.width << DELIM
     << loc.y + loc.height << DELIM
     << confidence << DELIM
     << COCO_LABELS[label] << DELIM
     << srcROIID_ << DELIM
     << ::md::str(origin) << DELIM
     << choiceOfBox;
  return ss.str();
}

} // namespace md
